core/domain/state_domain.py

```diff
@@ -745,16 +758,336 @@ def get_all_html_content_strings(self):
+        """
+        def convert_content_to_dict(ca_value, unused_obj_type):
+            """Conversion function used to convert SubtitledHtml to
+            SubtitledHtml dicts and SubtitledUnicode to SubtitledUnicode dicts.
+
+            Args:
+                ca_value: dict. Dictionary of key 'value' to
+                    original value of customization argument.
+                unused_obj_type: str. Indicates the obj_type found in
+                    the customization arguments schema.
+
+            Returns:
+                dict. The unmodified customization argument value.
+            """
+            return ca_value.to_dict()
+
+        return {
+            'value': InteractionCustomizationArg.traverse_by_schema_and_convert(
+                self.schema,
+                copy.deepcopy(self.value),
+                convert_content_to_dict
+            )
+        }
+
+    @classmethod
+    def from_customization_arg_dict(cls, ca_dict, ca_schema):
+        """Converts a customization argument dictionary to a
+        InteractionCustomizationArgument domain object. This is done by
+        traversing the customization argument schema, and converting
+        unicode to SubtitledUnicode and html to SubtitledHtml where appropriate.
+
+        Args:
+            ca_dict: dict. The customization argument dictionary. A dict of the
+                single key 'value' to the value of the customization argument.
+            ca_schema: dict. The schema that defines the customization argument
+                value.
+
+        Returns:
+            InteractionCustomizationArg. The customization argument domain
+            object.
+        """
+        def convert_content_to_domain_obj(ca_value, obj_type):
+            """Conversion function used to convert SubtitledHtml dicts to
+            SubtitledHtml and SubtitledUnicode dicts to SubtitledUnicode.
+
+            Args:
+                ca_value: dict. Dictionary of key 'value' to
+                    original value of customization argument.
+                obj_type: str. Indicates the obj_type found in
+                    the customization arguments schema.
+
+            Returns:
+                dict. The unmodified customization argument value.
+            """
+            if obj_type == schema_utils.SCHEMA_OBJ_TYPE_SUBTITLED_UNICODE:
+                return SubtitledUnicode(
+                    ca_value['content_id'], ca_value['unicode_str'])
+
+            if obj_type == schema_utils.SCHEMA_OBJ_TYPE_SUBTITLED_HTML:
+                return SubtitledHtml(
+                    ca_value['content_id'], ca_value['html'])
+
+        ca_value = InteractionCustomizationArg.traverse_by_schema_and_convert(
+            ca_schema,
+            copy.deepcopy(ca_dict['value']),
+            convert_content_to_domain_obj
+        )
+
+        return cls(ca_value, ca_schema)
+
+    def __init__(self, value, schema):
```

Comment:
This should be the method at the top, so that readers of this class know what the definition is.